source/support.tex

```diff
@@ -5668,6 +5677,45 @@
 of a template instantiation.
 \end{note}
 
+\rSec2[compare.type]{Type Ordering}
+
+\pnum
+There is an \impldef{total ordering of all types} total ordering of all types.
+For any (possibly incomplete) types \tcode{X} and \tcode{Y},
+the expression \tcode{\exposid{TYPE-ORDER}(X, Y)} is a constant expression\iref{expr.const}
+of type \tcode{strong_ordering}\iref{cmp.strongord}.
+Its value is \tcode{strong_ordering::less} if \tcode{X} precedes \tcode{Y}
+in this implementation-defined total order,
+\tcode{strong_ordering::greater} if \tcode{Y} precedes \tcode{X}, and
+\tcode{strong_ordering::equal} if they are the same type.
+\begin{note}
+\tcode{int}, \tcode{const int} and \tcode{int\&} are different types.
+\end{note}
+\begin{note}
+This ordering need not be consistent with the one induced by \tcode{type_info::before}.
+\end{note}
+\begin{note}
+The ordering of TU-local types from different translation units is not observable,
+because the necessary specialization of \tcode{type_order} is impossible to name.
+\end{note}
+
+\indexlibraryglobal{type_order}%
+\begin{itemdecl}
+template<class T, class U>
+  struct type_order;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+The name \tcode{type_order} denotes a \oldconcept{BinaryTypeTrait}\iref{meta.rqmts}
+with a base characteristic of
+\tcode{integral_constant<strong_ordering, \exposid{TYPE-ORDER}(T, U)>}.
```

Comment:
@AlisdairM can you fix the overfull hbox here?

If you sprinkle in `\brk{}` or a `\-` in the right place, surely that takes care of it.